interfaces/Portalicious/src/app/domains/user/user.api.service.ts

```diff
@@ -63,6 +63,53 @@ export class UserApiService extends DomainApiService {
   getAllUsers() {
     return this.generateQueryOptions<User[]>({
       path: [BASE_ENDPOINT],
+      queryKey: [BASE_ENDPOINT],
+    });
+  }
+
+  createUser({
+    username,
+    displayName,
+  }: {
+    username: string;
+    displayName: string;
+  }) {
+    return this.httpWrapperService.perform121ServiceRequest<User>({
+      method: 'POST',
+      endpoint: BASE_ENDPOINT,
+      body: {
+        users: [
+          {
+            username,
+            displayName,
+          },
+        ],
+      },
+    });
+  }
+
+  udpateUserDisplayName({
+    id,
+    displayName,
+  }: {
+    id: number | undefined;
+    displayName: string;
+  }) {
+    if (!id) {
+      return Promise.reject(new Error('User ID is required'));
+    }
+    return this.httpWrapperService.perform121ServiceRequest<User>({
+      method: 'PATCH',
+      endpoint: `${BASE_ENDPOINT}/${id.toString()}`,
+      body: {
+        displayName,
+      },
+    });
+  }
```

Comment:
Probably unnecessary, and I'll leave it to you whether to change this or not, but I feel like this might be more futureproof.

```suggestion

  udpateUser({
    id,
    patchedUser,
  }: {
    id: number | undefined;
    patchedUser: Partial<User>;
  }) {
    if (!id) {
      return Promise.reject(new Error('User ID is required'));
    }
    return this.httpWrapperService.perform121ServiceRequest<User>({
      method: 'PATCH',
      endpoint: `${BASE_ENDPOINT}/${id.toString()}`,
      body: patchedUser,
    });
  }
```